components/social/org.wso2.carbon.social.sql/src/main/java/org/wso2/carbon/social/sql/SQLActivityBrowser.java

```diff
@@ -119,9 +132,12 @@ public JsonObject getRating(String targetId) throws SocialActivityException {
 				resultSet.close();
 				if (total != 0) {
 					JsonObject object = new JsonObject();
-					object.addProperty(Constants.RATING, (double) total / count);
+					double average = (double) total / count;
+					object.addProperty(Constants.RATING, average);
 					object.addProperty(Constants.COUNT, count);
-					return object;
+					if (!Double.isInfinite(average)) {
+						return object;
+					}
```

Comment:
Thanks @daneshk   , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/8c713f553ab226a80c51662d9cc2813df4cab10b)